## After manual labelling different runs are stored in different .csv files here we merge these into one

In [6]:
import os

import pandas as pd

In [7]:
x = 0
for element in os.listdir('.'):
    if '.csv' in element:
        if x == 0:
            merge_df = pd.read_csv(element)
            x+=1
            continue
        current_df = pd.read_csv(element)
        merge_df = pd.concat([merge_df,current_df], ignore_index=True)

merge_df.head()

Unnamed: 0        1        2        3        4        5        6        7  \
0           0   1764.0   2224.0   2135.0   1864.0   1794.0   1727.0   1652.0   
1           1   1190.0   1781.0   1742.0   1476.0   1348.0   1278.0   5191.0   
2           2    667.0    803.0    788.0    734.0    732.0    677.0    679.0   
3           3  15369.0  16546.0  16678.0  16525.0  16585.0  16457.0  16479.0   
4           4    671.0    837.0    817.0    721.0    702.0    699.0    674.0   

         8        9      ...       \
0   1676.0   1693.0      ...        
1   1599.0   1187.0      ...        
2    671.0    665.0      ...        
3  16742.0  16759.0      ...        
4    674.0    661.0      ...        

                                            FileName  Element   Energy  \
0  EUV067_Ar_60mA_3p398kV_60s_1D15_5scook_2p44e-5...       18   3398.0   
1  EUV074_Ar_14p983k_140p1mA_5scook_60s_1D10_1p20...       18  14983.0   
2  EUV004_Ar_60mA_3p323kV_30s_1D6_5scook_2p59e-5t...       18   3323.0   
3  EUV001_150mA_3p98KV_60s_1D5_5scook_4p29e-6torr...        0   3980.0   
4  EUV006_Ar_60mA_3p353kV_30s_1D6_5scook_2p62e-5t...       18   3353.0   

   Current        Date  Note  Calibration1  Calibration2  Calibration3  \
0     60.0  11-18-2015   NaN           NaN           NaN           NaN   
1    140.1  11-19-2015   NaN           NaN           NaN           NaN   
2     60.0  11-19-2015   NaN           NaN           NaN           NaN   
3    150.0  11-17-2015   NaN           NaN           NaN           NaN   
4     60.0  11-19-2015   NaN           NaN           NaN           NaN   

   Calibration4  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4           NaN  

[5 rows x 2060 columns]

Preview after merge there are 5 columns with no values at all (Note,Calibration1,Calibration2,Calibration3,Calibration4) although we will need these in the future for now I'll take them out. Also forgot to tell pandas not to use index column so taking that out as well.

In [8]:
merge_df = merge_df.iloc[:,1:-5]
merge_df.head()

1        2        3        4        5        6        7        8  \
0   1764.0   2224.0   2135.0   1864.0   1794.0   1727.0   1652.0   1676.0   
1   1190.0   1781.0   1742.0   1476.0   1348.0   1278.0   5191.0   1599.0   
2    667.0    803.0    788.0    734.0    732.0    677.0    679.0    671.0   
3  15369.0  16546.0  16678.0  16525.0  16585.0  16457.0  16479.0  16742.0   
4    671.0    837.0    817.0    721.0    702.0    699.0    674.0    674.0   

         9       10     ...        2045    2046    2047    2048  \
0   1693.0   1698.0     ...      1568.0  1578.0  1587.0  1580.0   
1   1187.0   1161.0     ...      1052.0  1056.0  1067.0  1082.0   
2    665.0    671.0     ...       634.0   642.0   645.0   655.0   
3  16759.0  16748.0     ...      7185.0  6306.0  5595.0  4954.0   
4    661.0    675.0     ...       645.0   634.0   628.0   650.0   

      FolderName                                           FileName  Element  \
0  18_Ar_2015Nov  EUV067_Ar_60mA_3p398kV_60s_1D15_5scook_2p44e-5...       18   
1  18_Ar_2015Nov  EUV074_Ar_14p983k_140p1mA_5scook_60s_1D10_1p20...       18   
2  18_Ar_2015Nov  EUV004_Ar_60mA_3p323kV_30s_1D6_5scook_2p59e-5t...       18   
3  18_Ar_2015Nov  EUV001_150mA_3p98KV_60s_1D5_5scook_4p29e-6torr...        0   
4  18_Ar_2015Nov  EUV006_Ar_60mA_3p353kV_30s_1D6_5scook_2p62e-5t...       18   

    Energy  Current        Date  
0   3398.0     60.0  11-18-2015  
1  14983.0    140.1  11-19-2015  
2   3323.0     60.0  11-19-2015  
3   3980.0    150.0  11-17-2015  
4   3353.0     60.0  11-19-2015  

[5 rows x 2054 columns]

In [9]:
len(merge_df.Element[merge_df.Element==0])

962

For 962 files the code couldn't extract the element. Most of these are background but some are just formats the code couldn't read. Looping through these and adding them manually.

In [10]:
def checkZeros(df):
    
    for i, element in enumerate(df.Element):
        if element == 0:
            eid = input(df.FileName[i] + ': ')
            if eid == '':
                eid = 0
            else:
                eid = int(eid)
            df.Element.loc[i] = eid
            
    return df

In [11]:
merge_df = checkZeros(merge_df)

EUV001_150mA_3p98KV_60s_1D5_5scook_4p29e-6torr_11172015.SPE: 


/usr/local/lib/python3.6/site-packages/pandas/core/indexing.py:194: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._setitem_with_indexer(indexer, value)


EUV014_60mA_2p29KV_60s_1D3_5scook_2p75torr_11172015.SPE: 
EUV001_BG_150mA_3p98KV_60s_1D5_5scook_gateclosed_11172015.SPE: 
EUV011_60mA_2p19KV_60s_1D5_5scook_2p4e-6ttorr_11172015.SPE: 
EUV018_60mA_2p23KV_30s_1D6_5scook_2p42e-5torr_11172015.SPE: 
EUV012_60mA_2p29KV_60s_1D3_5scook_1p79e-5torr_11172015.SPE: 
EUV013_60mA_2p29KV_60s_1D3_5scook_3p38e-5torr_11172015.SPE: 
EUV003_150mA_3p98KV_60s_1D3_5scook_4p2e-6torr_11172015.SPE: 
EUV016_60mA_2p20KV_60s_1D3_5scook_2p52e-5torr_11172015.SPE: 
EUV005_60mA_2p29KV_60s_1D5_5scook_1p98e-6torr_11172015.SPE: 
EUV019_60mA_2p25KV_30s_1D6_5scook_2p55e-5torr_11172015.SPE: 
EUV002_150mA_3p98KV_60s_1D5_5scook_3p6e-6torr_11172015.SPE: 
EUV010_60mA_2p32KV_60s_1D5_5scook_4p24e-6ttorr_11172015.SPE: 
EUV006_60mA_2p34KV_60s_1D5_5scook_1p59e-6torr_11172015.SPE: 
EUV017_60mA_2p22KV_30s_1D6_5scook_2p45e-5torr_11172015.SPE: 
EUV009_60mA_2p28KV_60s_1D5_5scook_4p07e-6ttorr_11172015.SPE: 
EUV004_60mA_2p24KV_60s_1D5_5scook_3p37e-6torr_11172015.SPE: 
EUV007_60mA_2p30KV_60s

EUV351_Xe124_6k_150p0mA_60s_1D5_10scook_1p55E-5torr_06022016.SPE: 54
EUV440_Xe124_6k_150p0mA_60s_1D5_10scook_1p61e-5torr_06022016.SPE: 54
EUV670_NeXe136mix_6p0k_150p0mA_60s_1D5_10scook_4p83E-5torr_06042016.SPE: 
EUV524_Xe136_6k_150p0mA_60s_1D5_10scook_1p51e-5_06032016.SPE: 54
EUV491_Xe136_6k_150p0mA_60s_1D5_10scook_1p38e-5torr_06022016.SPE: 54
EUV279_Xe136_6k_150p0mA_60s_1D5_10scook_1p73E-5torr_06012016.SPE: 54
EUV253_Xe136_6k_150p0mA_60s_1D5r_10scook_1p51e-5torr_06012016.SPE: 54
EUV193_Xe136_6k_150p0mA_60s_1D5r_10scook_1p46E-5_06012016.SPE: 54
EUV142_Xe124_6k_150p0mA_60s_1D5_10scook_1p47E-5torr_06012016.SPE: 54
EUV658_NeXe124mix_6p0k_150p0mA_60s_1D5_10scook_4p38E-5torr_06042016.SPE: 
EUV544_Krypton_6k_150p0mA_60s_1D5_10scook_0p82e-5_06032016.SPE: 36
EUV043_Xe136_6k_150p0mA_60s_1D5__1p49e-05torr_10scook_05312016.SPE: 54
EUV124_Xe-136_6k_150p0mA_60s_1D5_10scook_1p46E-5torr_06012016.SPE: 54
EUV562_XeKrMix_6k_150p0mA_60s_1D5_10scook_0p77e-5_06032016.SPE: 
EUV621_NeXe124mix_6p0k_150p0mA_60

EUV520_Xe136_4p5k_150p0mA_60s_1D5_10scook_1p53e-5_06032016.SPE: 54
EUV047_Xe136_6k_150p0mA_60s_1D5__1p43e-05torr_10scook_05312016.SPE: 54
EUV283_BG_6k_150p0mA_60s_1D5_10scook06012016.SPE: 
EUV235_Xe124_6k_150p0mA_60s_1D5r_10scook_1p46e-5torr_06012016.SPE: 54
EUV543_Krypton_6k_150p0mA_60s_1D5_10scook_1p47e-5_06032016.SPE: 36
EUV578_XeKrMix_8k_150p0mA_20s_1D5_10scook_0p787e-5_06032016.SPE: 
EUV340_Xe136_6k_150p0mA_60s_1D5_10scook_1p47E-5torr_06022016.SPE: 54
EUV290_Xe124_6k_150p0mA_60s_1D5_10scook_1p52E-5torr_06012016.SPE: 54
EUV510_Xe136_4p2k_150p0mA_60s_1D5_10scook_1p55e-5_06032016.SPE: 54
EUV331_Xe136_6k_150p0mA_60s_1D5_10scook_1p51E-5torr_06022016.SPE: 54
EUV642_NeXe136mix_6p0k_150p0mA_60s_1D5_10scook_4p71E-5torr_06042016.SPE: 
EUV123_Xe-136_6k_150p0mA_60s_1D5_10scook_1p47E-5torr_06012016.SPE: 54
EUV201_Xe124_6k_150p0mA_60s_1D5r_10scook_1p53E-5_06012016.SPE: 54
EUV311_Xe136_6k_150p0mA_60s_1D5_10scook_1p44E-5torr_06022016.SPE: 54
EUV038_Xe136_6k_150p0mA_60s_1D5__1p58e-05torr_10scook_0

EUV231_Xe124_6k_150p0mA_60s_1D5r_10scook_1p48e-5torr_06012016.SPE: 54
EUV252_Xe136_6k_150p0mA_60s_1D5r_10scook_1p48e-5torr_06012016.SPE: 54
EUV323_Xe124_6k_150p0mA_60s_1D5_10scook_1p46E-5torr_06022016.SPE: 54
EUV343_Xe136_6k_150p0mA_60s_1D5_10scook_1p51E-5torr_06022016.SPE: 54
EUV479_Xe136_6k_150p0mA_60s_1D5_10scook_1p43e-5torr_06022016.SPE: 54
EUV653_NeXe124mix_6p0k_150p0mA_60s_1D5_10scook_4p60E-5torr_06042016.SPE: 
EUV086_Xe-124_6k_150p0mA_60s_1D5_1p57e-5torr_10scook_06012016.SPE: 54
EUV150_Xe136_6k_150p0mA_60s_1D5_10scook__1p44E-5torr_06012016.SPE: 54
EUV394_Xe136_6k_150p0mA_60s_1D5_10scook_1p68E-5torr_06022016.SPE: 54
EUV179_Xe124_6k_150p0mA_60s_1D5__1p51e-05torr_10scook_06012016.SPE: 54
EUV094_Xe-124_6k_150p0mA_60s_1D5_1p36e-5torr_10scook_06012016.SPE: 54
EUV389_Xe124_6k_150p0mA_60s_1D5_10scook_1p54E-5torr_06022016.SPE: 54
EUV493_BG_6k_150p0mA_60s_1D5_10scook_06022016.SPE: 
EUV314_BG_6k_150p0mA_60s_1D5_10scook_06022016.SPE: 
EUV584_XeKrMix_6k_150p0mA_60s_1D5_10scook_1p06e-5_060320

EUV059_Xe124_6k_150p0mA_60s_1D5_10scook_1p44E-5torr_05312016.SPE: 54
EUV041_Xe136_6k_150p0mA_60s_1D5__1p55e-05torr_10scook_05312016.SPE: 54
EUV215_Xe136_6k_150p0mA_60s_1D5r_10scook_1p48E-5_06012016.SPE: 54
EUV422_Xe124_6k_150p0mA_60s_1D5_10scook_1p52E-5torr_06022016.SPE: 54
EUV497_Xe124_6k_150p0mA_60s_1D5_10scook_1p30e-5torr_06022016.SPE: 54
EUV145_Xe124_6k_150p0mA_60s_1D5_10scook_1p43E-5torr_06012016.SPE: 54
EUV200_Xe124_6k_150p0mA_60s_1D5r_10scook_1p49E-5_06012016.SPE: 54
EUV008_Xe136_6k_150p0mA_60s_1D5__10scook_1p50e-5torr_05312016.SPE: 54
EUV287_Xe124_6k_150p0mA_60s_1D5_10scook_1p51E-5torr_06012016.SPE: 54
EUV569_XeKrMix_6k_150p0mA_60s_1D5_10scook_0p89e-5_06032016.SPE: 
EUV525_Xe136_6k_150p0mA_60s_1D5_10scook_1p49e-5_06032016.SPE: 54
EUV338_Xe136_6k_150p0mA_60s_1D5_10scook_1p46E-5torr_06022016.SPE: 54
EUV539_background_6k_150p0mA_60s_1D5_10scook_1p46e-5_06032016.SPE: 
EUV057_Xe124_6k_150p0mA_60s_1D5_10scook_1p44E-5torr_05312016.SPE: 54
EUV132_Xe-136_6k_150p0mA_60s_1D5_10scook_1p68E

EUV349_Xe124_6k_150p0mA_60s_1D5_10scook_1p56E-5torr_06022016.SPE: 54
EUV073_Xe-136_6k_150p0mA_60s_1D5_1p51e-5_10scook_06012016.SPE: 54
EUV036_Xe136_6k_150p0mA_60s_1D5__1p52e-05torr_10scook_05312016.SPE: 54
EUV466_Xe124_6k_150p0mA_60s_1D5_10scook_1p36e-5torr_06022016.SPE: 54
EUV013_Xe136_6k_150p0mA_60s_1D5__10scook_1p52e-5torr_05312016.SPE: 54
EUV274_Xe136_6k_150p0mA_60s_1D5_10scook_1p53E-5torr_06012016.SPE: 54
EUV083_Xe-124_6k_150p0mA_60s_1D5_1p43e-5torr_10scook_06012016.SPE: 54
EUV079_Xe-136_6k_150p0mA_60s_1D5_1p6e-5_10scook_06012016.SPE: 54
EUV431_Xe136_6k_150p0mA_60s_1D5_10scook_1p56E-5torr_06022016.SPE: 54
EUV011_Xe136_6k_150p0mA_60s_1D5__10scook_1p47e-5torr_05312016.SPE: 54
EUV411_Xe124_6k_150p0mA_60s_1D5_10scook_1p51E-5torr_06022016.SPE: 54
EUV276_Xe136_6k_150p0mA_60s_1D5_10scook_1p45E-5torr_06012016.SPE: 54
EUV662_NeXe136mix_6p0k_150p0mA_60s_1D5_10scook_5E-5torr_06042016.SPE: 
EUV664_NeXe136mix_6p0k_150p0mA_60s_1D5_10scook_5p08E-5torr_06042016.SPE: 
EUV402_Xe136_6k_150p0mA_60s_1

EUV050_BG_1520eV_1D5_35p7mA_CO2_3p4e-5_10s.SPE: 
EUV040_BG_5800eV_1D10_125mA_nogas_10s.SPE: 
EUV001_BG_6000eV_1D10_150mA_nogas_10sl.SPE: 
EUV018_Gd+Ne_3986eV_1D5_100mA_1e-4_10s.SPE: 
EUV015_CO2_1p8keV_57p3mA_10sdumping_1D10_60s_2p7e-5_9Jan2014.SPE: 
EUV027_CO2_2p03keV_60p1mA_10sdumping_1D10_60s_1p0e-4_9Jan2014.SPE: 
EUV016_bkg_1p65keV_52p5mA_10sdumping_1D10_60s_9Jan2014.SPE: 
EUV028_bkg_5p83keV_137p8mA_10sdumping_1D10_60s_9Jan2014.SPE: 
EUV094_150p0mA_1D10_4p85KV_60s_nodump_nogas_09182015.SPE: 
EUV095_150p0mA_1D10_4p85KV_60s_nodump_nogas_09182015.SPE: 
EUV099_150p1mA_1D10_6p97KV_60s_nodump_nogas_09182015.SPE: 
EUV097_147p1mA_1D10_3p98KV_60s_nodump_nogas_09182015.SPE: 
EUV061_140p1mA_1D10_8p00KV_60s_10sdump_nogas_09172015.SPE: 
EUV092_150p2mA_1D10_7p99KV_60s_nodump_nogas_09182015.SPE: 
EUV098_150p0mA_1D10_4p96KV_60s_nodump_nogas_09182015.SPE: 
EUV093_150p2mA_1D10_7p99KV_60s_nodump_nogas_09182015.SPE: 
EUV070_150p0mA_1D10_7p99KV_60s_10sdump_Mo+Ar_09172015.SPE: 
EUV096_150p0mA_1D10_5p98KV

r045_BG_E5495_1D5m_slit255_32-172_051811.SPE: 
r085_BG_E2005_1D5m_slit255_32-172_051911.SPE: 
r088_CO2_E2005_1D10m_slit255_32-172_051911.SPE: 
EUV015_CO2_1p8keV_57p3mA_10sdumping_1D10_60s_2p7e-5_9Jan2014.SPE: 
EUV027_CO2_2p03keV_60p1mA_10sdumping_1D10_60s_1p0e-4_9Jan2014.SPE: 
EUV016_bkg_1p65keV_52p5mA_10sdumping_1D10_60s_9Jan2014.SPE: 
EUV028_bkg_5p83keV_137p8mA_10sdumping_1D10_60s_9Jan2014.SPE: 
EUV011_CO2_nogas_3600eV_4s_150ma.SPE: 
EUV012_BG_nogas_3600eV_4s_150ma.SPE: 
EUV006_BG_nogas_836eV_4s_9ma.SPE: 
EUV011_BG_20keV_145mA_30s_1D6_5scook_03132017_2.SPE: 
EUV011_BG_20keV_145mA_30s_1D6_5scook_03132017_1.SPE: 
EUV011_BG_20keV_145mA_30s_1D6_5scook_03132017_.SPE: 
EUV001_BG_8keV_150mA_s_1D5_nodump_11182016_2.SPE: 
EUV0114_BG_4p85keV_150mA_60s_1D5_nodump_11192016.SPE: 
EUV001_BG_8keV_150mA_s_1D5_nodump_11182016_1.SPE: 
EUV0111_BG_4p85keV_150mA_60s_1D5_5_nodump_11192016.SPE: 
EUV009_none_10p0keV_150mA_30s_1D5_5scook_11182016.SPE: 
EUV001_BG_8keV_150mA_s_1D5_nodump_11182016.SPE: 
EUV001_

In [12]:
len(merge_df.Element[merge_df.Element==0])

434

Looping through again just to double check.

In [20]:
merge_df = checkZeros(merge_df)

EUV001_150mA_3p98KV_60s_1D5_5scook_4p29e-6torr_11172015.SPE: 


/usr/local/lib/python3.6/site-packages/pandas/core/indexing.py:194: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._setitem_with_indexer(indexer, value)


EUV014_60mA_2p29KV_60s_1D3_5scook_2p75torr_11172015.SPE: 
EUV001_BG_150mA_3p98KV_60s_1D5_5scook_gateclosed_11172015.SPE: 
EUV011_60mA_2p19KV_60s_1D5_5scook_2p4e-6ttorr_11172015.SPE: 
EUV018_60mA_2p23KV_30s_1D6_5scook_2p42e-5torr_11172015.SPE: 
EUV012_60mA_2p29KV_60s_1D3_5scook_1p79e-5torr_11172015.SPE: 
EUV013_60mA_2p29KV_60s_1D3_5scook_3p38e-5torr_11172015.SPE: 
EUV003_150mA_3p98KV_60s_1D3_5scook_4p2e-6torr_11172015.SPE: 
EUV016_60mA_2p20KV_60s_1D3_5scook_2p52e-5torr_11172015.SPE: 
EUV005_60mA_2p29KV_60s_1D5_5scook_1p98e-6torr_11172015.SPE: 
EUV019_60mA_2p25KV_30s_1D6_5scook_2p55e-5torr_11172015.SPE: 
EUV002_150mA_3p98KV_60s_1D5_5scook_3p6e-6torr_11172015.SPE: 
EUV010_60mA_2p32KV_60s_1D5_5scook_4p24e-6ttorr_11172015.SPE: 
EUV006_60mA_2p34KV_60s_1D5_5scook_1p59e-6torr_11172015.SPE: 
EUV017_60mA_2p22KV_30s_1D6_5scook_2p45e-5torr_11172015.SPE: 
EUV009_60mA_2p28KV_60s_1D5_5scook_4p07e-6ttorr_11172015.SPE: 
EUV004_60mA_2p24KV_60s_1D5_5scook_3p37e-6torr_11172015.SPE: 
EUV007_60mA_2p30KV_60s

EUV589_KrXe124mix_6k_150p0mA_60s_1D5_10scook_0E-988torr_06032016.SPE: 
EUV257_BG_6k_150p0mA_60s_1D5r_10scook_06012016.SPE: 
EUV596_KrXe124mix_6k_150p0mA_60s_1D5_10scook_0E988torr_06032016.SPE: 
EUV184_BG_6k_150p0mA_60s_1D5r_10scook_06012016.SPE: 
EUV167_BG_6k_150p0mA_60s_1D5__1p44e-05torr_10scook_06012016.SPE: 
EUV551_KrXeMix_6k_150p0mA_60s_1D5_10scook_1p44e-5_06032016.SPE: 
EUV579_XeKrMix_6k_150p0mA_20s_1D5_10scook_0p787e-5_06032016.SPE: 
EUV654_NeXe124mix_6p0k_150p0mA_60s_1D5_10scook_4p42E-5torr_06042016.SPE: 
EUV566_XeKrMix_6k_150p0mA_60s_1D5_10scook_0p95e-5_06032016.SPE: 
EUV064_1_BG_6k_150p0mA_60s_1D5_10scook_05312016.SPE: 
EUV629_BG_6p0k_150p0mA_60s_1D5_10scook_06042016.SPE: 
EUV578_XeKrMix_8k_150p0mA_60s_1D5_10scook_0p787e-5_06032016.SPE: 
EUV019_Bg_6k_150p0mA_60s_1D5__10scook_05312016.SPE: 
EUV585_XeKrMix_6k_150p0mA_60s_1D5_10scook_1p05e-5_06032016.SPE: 
EUV549_Krypton_6k_150p0mA_60s_1D5_10scook_0p73e-5_06032016.SPE: 36
EUV671_NeXe136mix_6p0k_150p0mA_60s_1D5_10scook_5p21E-5torr

r067_BG_E5700_1D5m_slit255_32-172_051911.SPE: 
r011_BG_E9015_1D5m_slit255_32-172_051711.SPE: 
r072_BG_E4000_1D5m_slit255_32-172_051911.SPE: 
r085_BG_E2005_1D5m_slit255_32-172_051911.SPE: 
r074_BG_E3600_1D5m_slit255_32-172_051911.SPE: 
r016_BG_E7015_1D5m_slit255_32-172_051711.SPE: 
r081_BG_E2755_1D5m_slit255_32-172_051911.SPE: 
r088_CO2_E2005_1D10m_slit255_32-172_051911.SPE: 
r078_BG_E3600_1D1m_FM_slit255_32-172_051911.SPE: 
r020_BG_E6015_1D5m_slit255_32-172_051711.SPE: 
r026_BG_E2035_1D5m_slit255_32-172_051711.SPE: 
r012_BG_E8015_1D5m_slit255_32-172_051711.SPE: 
EUV005_BKG_2p2keV_66p2mA_10sdumping_nogas_1D10_60s_26Jul2013.SPE: 
EUV012_BKG_1p16keV_18p3mA_10sdumping_nogas_1D10_60s_26Jul2013.SPE: 
EUV019_BKG_p98keV_15p2mA_10sdumping_nogas_1D5_60s_26Jul2013.SPE: 
EUV011_BKG_1p20keV_19p3mA_10sdumping_nogas_1D10_60s_26Jul2013.SPE: 
EUV045_65p1mA_1D10_2p23KV_60s_10sdump_CO2_09162015.SPE: 
EUV044_147p0mA_1D10_8p07KV_60s_10sdump_CO2_09162015.SPE: 
EUV059_150p1mA_1D10_4p85KV_60s_nodump_nogas_091

EUV019_BG_9p5keV_130mA_60s_1D5_5scook_08022017.SPE: 
EUV022_BG_9p5keV_130mA_60s_1D5_5scook_08022017.SPE: 
EUV012_BG_5p2keV_132p1mA_60s_1D5_5scook_08022017.SPE: 
EUV088_BG_9p81keV_140mA_60s_1D5_5scook_08022017_1.SPE: 
EUV023_BG_9p0keV_130mA_60s_1D5_5scook_08022017_1.SPE: 
EUV078_BG_9p29keV_140mA_60s_1D5_5scook_08022017_1.SPE: 
EUV073_BG_7p8keV_140mA_60s_1D5_5scook_08022017.SPE: 
EUV102_BG_7p8keV_140p0mA_60s_1D5_5scook_08042017_ 1.SPE: 
EUV077_BG_9p2keV_140mA_60s_1D5_5scook_08022017_1.SPE: 
EUV100_BG_7p9keV_140p0mA_60s_1D5_5scook_08042017_1.SPE: 
EUV092_BG_9p99keV_140mA_60s_1D5_5scook_08022017.SPE: 
EUV046_Kr9xxxeV_140UDT_1D5_140mA_3p5em5torr_18May2012.SPE: 36
EUV050_BG_6305eV_1D5_150mA_Ar1p6e-5_4sl.SPE: 
EUV001_nomevva_28740eV_1D5_150mA_nogas_10sl.SPE: 
EUV002_nomevva_28740eV_1D5_150mA_nogas_10sl.SPE: 
EUV072_BG_6266eV_1D10_100mA_nogas_10sl.SPE: 
EUV059_BG_6208eV_1D10_100mA_nogas_10sl.SPE: 


In [29]:
len(merge_df.Element[merge_df.Element==0])

427

In [30]:
merge_df.to_csv('specBase.csv', index=None)